### PR TITLE
Emulate the parts of FontAwesome that we use

### PR DESCRIPTION
**Pull request by ppalaga**
Motivated by https://github.com/apache/camel-quarkus/issues/1410 , based on https://gitlab.com/antora/antora-ui-default/-/merge_requests/121/diffs

### CODE REVIEW

**Comment by zregvart** [comment]
I suspect that this will add significant weight to the pages, and I'm not sure if we're going to use more than several FontAwesome icons. Perhaps it would be worthwhile to investigate a [customized version](https://github.com/FortAwesome/Font-Awesome/wiki/Customize-Font-Awesome#the-easy-way) including only icons that we do use. As an alternative we could use the unicode lock symbol: `&#x1f512;` (&#x1f512;).

File: antora-ui-camel/gulp.d/tasks/build.js [hunk 1]
```diff
@@ -46,7 +46,18 @@ module.exports = (src, dest, preview) => () => {
           const basename = ospath.basename(abspath)
           const destpath = ospath.join(dest, 'font', basename)
           if (!fs.pathExistsSync(destpath)) fs.copySync(abspath, destpath)
-          return path.join('..', 'font', basename)
+          return path.join('..', 'font', basename) + asset.search + asset.hash
```
Comment (zregvart) [inline antora-ui-camel/gulp.d/tasks/build.js:49]: Not sure about this change, doesn't seem to be related to adding Font Awesome.

**Comment by ppalaga** [comment]
Thanks for the feedback, @zregvart ! I am a frontend noob happy to follow any advice :) OK, let me try to extract only the relevant CSS and try the unicode glyph.

**Comment by ppalaga** [comment]
c0e2e6b adopts the proposal to emulate `fa-lock` with 🔒 unicode character

**Comment by ppalaga** [comment]
ups, there are changes in c0e2e6b that should not be there, let me filter them out

**Comment by ppalaga** [comment]
f25b8f5 is what it should be

**Comment by ppalaga** [comment]
The lock sign can be seen e.g. here https://deploy-preview-457--camel.netlify.app/camel-quarkus/latest/extensions/microprofile-metrics.html#_additional_camel_quarkus_configuration 
Would please somebody review?

**Comment by zregvart** [comment]
@ppalaga this can be done without adding any new CSS, i.e. by adding the icon as [unicode character](https://docs.antora.org/antora/2.3/asciidoc/special-characters-and-symbols/) to the adoc file. I think that way we would not couple what icons can be used in docs to the state of CSS on the website.

**Comment by ppalaga** [comment]
> @ppalaga this can be done without adding any new CSS, i.e. by adding the icon as [unicode character]

In our situation, it is rather hard to do, because we get `icon:lock[title=Fixed at build time]` hard coded from Quarkus tooling https://github.com/quarkusio/quarkus/blob/master/core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java#L78 and I'd prefer not to rewrite it.

I thought, because font awesome is commonly used in both AsciiDoc [1][2] and Antora (it is a part of the default UI now [3]), having some level of CSS support on our website would be quite natural too. 

[1] https://asciidoctor.org/docs/user-manual/#admonition-icons
[2] https://asciidoctor.org/docs/user-manual/#inline-icons
[3] https://gitlab.com/antora/antora-ui-default/-/merge_requests/121

**Comment by zregvart** [comment]
> > 
> 
> In our situation, it is rather hard to do, because we get `icon:lock[title=Fixed at build time]` hard coded from Quarkus tooling [https://github.com/quarkusio/quarkus/blob/master/core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java#L78](https://github.com/quarkusio/quarkus/blob/master/core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java?rgh-link-date=2020-08-18T08%3A27%3A36Z#L78) and I'd prefer not to rewrite it.

Right, I thought we were in full control of the docs.
